### PR TITLE
tries to use underlying nats connection for status

### PR DESCRIPTION
**Pull request by thinktainer**

### CODE REVIEW

File: proximo-server/main.go [hunk 1]
```diff
@@ -100,10 +102,12 @@ func main() {
 			EnvVar: "PROXIMO_NATS_CLUSTER_ID",
 		})
 		cmd.Action = func() {
-			cr.handler = &natsStreamingHandler{
-				url:       *url,
-				clusterID: *cid,
+			nh, err := NewNatsStreamingHandler(*url, *cid)
+			if err != nil {
+				log.Fatal(err)
```
Comment (jakekeeys) [inline proximo-server/main.go]: Panic

File: proximo-server/nats-streaming.go [hunk 2]
```diff
@@ -14,17 +14,29 @@ import (
 )
 
 type natsStreamingHandler struct {
-	url       string
 	clusterID string
+	nc        *nats.Conn
+}
+
+func NewNatsStreamingHandler(url, clusterID string) (*natsStreamingHandler, error) {
+	nc, err := nats.Connect(url, nats.Name("proximo-nats-streaming-"+generateID()))
```
Comment (jakekeeys) [inline proximo-server/nats-streaming.go]: nats.Name?

Comment (thinktainer) [reply proximo-server/nats-streaming.go]: that's what the option is called, it modifies the options object, it's the functional ...optionFunc pattern

File: proximo-server/nats-streaming.go [hunk 3]
```diff
@@ -14,17 +14,29 @@ import (
 )
 
 type natsStreamingHandler struct {
-	url       string
 	clusterID string
+	nc        *nats.Conn
+}
+
+func NewNatsStreamingHandler(url, clusterID string) (*natsStreamingHandler, error) {
```
Comment (mjgarton) [inline proximo-server/nats-streaming.go]: This doesn't need to be exported.

**Review by mjgarton — APPROVED** [review]

**Pull request merged**